server/database_fns.py

```diff
@@ -26,117 +28,51 @@ def real_dicts_to_python_dicts(real_dict_list):
     return real_dict_list
 
 
-def get_all_queued():
-    """
-
-    Get all queued jobs from the DB, sorted by
-    earliest submit_time. Returns them as a list of dicts.
-
-    """
-
-    command = """
-                SELECT
-                    *
-                FROM
-                    jobs
-                WHERE
-                    status = 'QUEUED'
-                ORDER BY
-                    submit_time ASC;
-                """
-
-    conn = None
-    try:
-
-        conn = psycopg2.connect(**DB_KWARGS)  # Connect to DB
-        cur = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)  # Get cursor
-        cur.execute(command)  # Send the command
-
-        # Returns just the top one. If there is no queued, this
-        # value should be None.
-        rows = cur.fetchall()
-
-        cur.close()  # close communication with the PostgreSQL database server
-        conn.commit()  # commit the changes
-
-        return real_dicts_to_python_dicts(rows)
-
-    except (Exception, psycopg2.DatabaseError) as error:
-        print(error)
-    finally:
-        if conn is not None:
-            conn.close()
-
-
-def get_all_running():
+def get_all_jobs_by_status(status):
     """
 
-    Get all running jobs from the DB, sorted by
-    earliest submit_time. Returns them as a list of dicts.
-
-    """
-
-    command = """
-                SELECT
-                    *
-                FROM
-                    jobs
-                WHERE
-                    status = 'RUNNING'
-                ORDER BY
-                    start_time ASC;
-                """
-
-    conn = None
-    try:
-
-        conn = psycopg2.connect(**DB_KWARGS)  # Connect to DB
-        cur = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)  # Get cursor
-        cur.execute(command)  # Send the command
-
-        # Returns just the top one. If there is no queued, this
-        # value should be None.
-        rows = cur.fetchall()
-
-        cur.close()  # close communication with the PostgreSQL database server
-        conn.commit()  # commit the changes
-
-        return real_dicts_to_python_dicts(rows)
-
-    except (Exception, psycopg2.DatabaseError) as error:
-        print(error)
-    finally:
-        if conn is not None:
-            conn.close()
-
-
-def get_all_completed():
-    """
-
-    Get all completed jobs from the DB, sorted by
-    earliest submit_time. Returns them as a list of dicts.
+    Get all jobs of one type from the DB, sorted by earliest submit_time.
+    Returns them as a list of dicts.
 
     """
 
+    assert (
+        status in VALID_STATUSES
+    ), f"Must provide a valid status! Provided = {status}"
+
+    # For COMPLETED, only want the most recent
+    status_sort_order = {
+        "QUEUED": "ASC",
+        "RUNNING": "ASC",
+        "COMPLETED": "DESC",
+        "FAILED": "DESC",
+    }
```

Comment:
Can these get passed into the function with a default rather than this special dict inside the function. Same for n_lim. Then maybe a docstring to splain these.

`get_all_jobs_by_status(status)` -> get_all_jobs_by_status(status, order="DESC", limit=20)

Reply:
I think it's simpler as is.